Plugin.php

```diff
@@ -366,4 +373,56 @@ public function translateRawPlural($string, $count = 0, $params = [], $locale =
     {
         return Lang::choice(Message::transRaw($string, $params, $locale), $count, $params);
     }
+
+    public function extendWinterSitemap()
+    {
+        Event::listen('winter.sitemap.processMenuItems', function ($item, $url, $theme, $apiResult) {
+            if ($item->type === 'cms-page') {
+                return Classes\MLCmsPage::resolveMenuItem($item, $url, $theme);
+            }
+
+            if (PluginManager::instance()->exists('Winter.Pages')) {
+                if ($item->type === 'static-page') {
+                    return Classes\MLStaticPage::resolveMenuItem($item, $url, $theme);
+                }
+            }
+
+            if (PluginManager::instance()->exists('Winter.blog')) {
+                switch ($item->type) {
+                    case 'blog-category':
+                    case 'all-blog-categories':
+                        return Classes\MLBlogCategoryModel::resolveMenuItem($item, $url, $theme);
+                    case 'blog-post':
+                    case 'category-blog-posts':
+                    case 'all-blog-posts':
+                        return Classes\MLBlogPostModel::resolveMenuItem($item, $url, $theme);
+                }
+            }
+            return false;
```

Comment:
```suggestion
            switch ($item->type) {
                case 'cms-page':
                    return Classes\MLCmsPage::resolveMenuItem($item, $url, $theme);
                case 'static-page':
                    return Classes\MLStaticPage::resolveMenuItem($item, $url, $theme);
                case 'blog-category':
                case 'all-blog-categories':
                    return Classes\MLBlogCategoryModel::resolveMenuItem($item, $url, $theme);
                case 'blog-post':
                case 'category-blog-posts':
                case 'all-blog-posts':
                    return Classes\MLBlogPostModel::resolveMenuItem($item, $url, $theme);
                default:
                    return false;
            }
```

Reply:
This was done on purpose for when those plugins are NOT installed/enabled.

Reply:
Sure, but if they're not installed / enabled then they won't have entries in the sitemap.

Reply:
Are you playing devil's advocate here? lol

There may be cases where left-over menuItemTypes from the blog plugin are in the Sitemap, and the blog plugin has been disabled/uninstalled without getting rid of these items... in which case the sitemap would silently (since this can easily be overlooked) break. With these conditional blocks in place, the sitemap would keep working.